hydromt/raster.py

```diff
@@ -849,62 +966,73 @@ def clip_bbox(self, bbox, align=None, buffer=0):
         buffer : int, optional
             Buffer around the bounding box expressed in resolution multiplicity,
             by default 0
+        crs : CRS, int, str, optional
+            crs of bbox
 
         Returns
         -------
         xarray.DataSet or DataArray
             Data clipped to bbox
         """
+        if crs is not None:
+            if not isinstance(crs, pyproj.CRS):
+                crs = pyproj.CRS.from_user_input(crs)
+            if crs != self.crs:
+                bbox = rasterio.warp.transform_bounds(crs, self.crs, *bbox)
         w, s, e, n = bbox
-        left, bottom, right, top = self.internal_bounds
-        xres, yres = [np.abs(res) for res in self.res]
+
         if align is not None:
             align = abs(align)
             # align to grid
             w = (w // align) * align
             s = (s // align) * align
             e = (e // align + 1) * align
             n = (n // align + 1) * align
-        if top > bottom:
-            n = min(top, n + buffer * yres)
-            s = max(bottom, s - buffer * yres)
-            y_slice = slice(n, s)
-        else:
-            n = min(bottom, n + buffer * yres)
-            s = max(top, s - buffer * yres)
-            y_slice = slice(s, n)
-        if left > right:
-            e = min(left, e + buffer * xres)
-            w = max(right, w - buffer * xres)
-            x_slice = slice(e, w)
+        xres, yres = self.res
+        y0, y1 = (n, s) if yres < 0 else (s, n)
+        x0, x1 = (e, w) if xres < 0 else (w, e)
+        if buffer > 0:
+            y0 -= yres * buffer
+            y1 += yres * buffer
+            x0 -= xres * buffer
+            x1 += xres * buffer
+        if self.rotation != 0:
+            # NOTE not sure what is expected when clipping a rotated grid with a bbox
+            # this keeps the largest grid where all row / col have at least one cell inside the bbox
+            w, e = min(x0, x1), max(x0, x1)
+            s, n = min(y0, y1), max(y0, y1)
+            gdf_mask = gpd.GeoDataFrame(geometry=[box(w, s, e, n)], crs=self.crs)
+            da_mask = self.geometry_mask(gdf_mask)
+            return self.clip_mask(da_mask)
         else:
-            e = min(right, e + buffer * xres)
-            w = max(left, w - buffer * xres)
-            x_slice = slice(w, e)
-        return self._obj.sel({self.x_dim: x_slice, self.y_dim: y_slice})
+            return self._obj.sel({self.x_dim: slice(x0, x1), self.y_dim: slice(y0, y1)})
 
-    def clip_mask(self, mask):
+    def clip_mask(self, da_mask: xr.DataArray, mask: bool = False):
```

Comment:
I've tried to make the arguments of the `clip_mask` and `clip_geoms` functions consistent. Now the mask argument means different things in both methods. Don't think this can be done without breaking things, but it's easily fixed.

I would actually like to also remove the setting of a _mask_ variable in the coords which is unclear but might also break things downstream.

Reply:
Not nice that it breaks though, we need to coordinate the release of the plugins as well asap (for wflow in breaks already in the setup_basemaps method). I agree better if things are more consistent though so not an easy choice.

For the mask, I know for sure removing it will break a lot of things downstream. I think having a mask variable in the coordinates to mask non-active cells of your grid can be quite useful, so I am not sure if we should remove it or feature it better within hydromt.

Reply:
I reverted this back for now and created a new issue https://github.com/Deltares/hydromt/issues/284